### PR TITLE
Reload

### PR DESCRIPTION
**Pull request by jimmywutangclan**
- Added ammo limits for players to shoot
- Added reloading to reset ammo count
- Added reload animation to GunController
- Added reload sound effect

### CODE REVIEW

File: VolleyPaint/Assets/Blobcreate/Projectile Toolkit/Core/TrajectoryPredictor.cs [hunk 1]
```diff
@@ -14,11 +14,10 @@ public class TrajectoryPredictor : MonoBehaviour
...
 		public int PositionCount => validCount;
-
-		/// <summary>
-		/// The gravity that used to simulate trajectories. Currently does not support custom gravity direction.
-		/// </summary>
-		public Vector3 Gravity
+        /// <summary>
+        /// The gravity that used to simulate trajectories. Currently does not support custom gravity direction.
+        /// </summary>
+        public Vector3 Gravity
```
Comment (slevenstein) [inline VolleyPaint/Assets/Blobcreate/Projectile Toolkit/Core/TrajectoryPredictor.cs]: the spacing seems messed up here, you might want to change it back

File: VolleyPaint/Assets/Scripts/Ability/AbilityHolder.cs [hunk 2]
```diff
@@ -2,9 +2,11 @@
 using System.Collections.Generic;
 using UnityEngine;
 using UnityEngine.Events;
+using Unity.Netcode;
 
-public class AbilityHolder : MonoBehaviour
+public class AbilityHolder : NetworkBehaviour
```
Comment (slevenstein) [inline VolleyPaint/Assets/Scripts/Ability/AbilityHolder.cs]: these seem like Kims changes so we should probably merge all his changes from the ability test branch first

File: VolleyPaint/Assets/Scripts/Game/TeamAssignment.cs [hunk 3]
```diff
@@ -15,8 +15,7 @@ void Start()
         assignedTeam = GameObject.Find("GameManager").GetComponent<GameManagement>().GetTeamToAutoAssignTo();
 
         // turn off character controller before teleporting
-        CharacterController cc = gameObject.GetComponent<CharacterController>();
```
Comment (slevenstein) [inline VolleyPaint/Assets/Scripts/Game/TeamAssignment.cs]: what is this change for?

Comment (slevenstein) [reply VolleyPaint/Assets/Scripts/Game/TeamAssignment.cs]: not sure if this is kims changes

File: VolleyPaint/Assets/Scripts/Player/Shooting/PlayerShoot.cs [hunk 4]
```diff
@@ -65,7 +70,30 @@ void Update()
...
+            // End reloading
+            if (reloadTimeElapsed >= reloadCooldown)
+            {
+                isReloading = false;
+                reloadSoundPlayed = false;
+                reloadTimeElapsed = 0.0f;
+                bulletsLeft = bulletLimit;         
+            }
+        }
+        if (!isReloading && Input.GetButton("Fire1") && fireRateCountDown >= fireRate)
```
Comment (slevenstein) [inline VolleyPaint/Assets/Scripts/Player/Shooting/PlayerShoot.cs]: nit pick but u could do an else if here and remove the !isReloading

Comment (slevenstein) [reply VolleyPaint/Assets/Scripts/Player/Shooting/PlayerShoot.cs]: or do something like this:

else {
   if(Input.GetButton("Fire1") && fireRateCountDown >= fireRate){
       ...
   }
   if(Input.GetKey(KeyCode.R) && bulletsLeft != bulletLimit){
      ...
   }
}

File: VolleyPaint/Assets/Scripts/Player/Shooting/PlayerShoot.cs [hunk 5]
```diff
@@ -188,6 +227,5 @@ private bool IsBallShot()
     public void ReplenishAmmoClientRpc()
```
Comment (slevenstein) [inline VolleyPaint/Assets/Scripts/Player/Shooting/PlayerShoot.cs]: hmm i dont think other players need to know about when u replenish ammo so u shouldn't need an RPC for this

File: VolleyPaint/Assets/Scripts/Game/TeamAssignment.cs (hunk 3 above)
Comment (jimmywutangclan) [reply VolleyPaint/Assets/Scripts/Game/TeamAssignment.cs]: I think those are Kim's, yeah.

File: VolleyPaint/Assets/Scripts/Ability/AbilityHolder.cs (hunk 2 above)
Comment (jimmywutangclan) [reply VolleyPaint/Assets/Scripts/Ability/AbilityHolder.cs]: I think those are Kim's, yeah.

File: VolleyPaint/Assets/Scripts/Player/Shooting/PlayerShoot.cs (hunk 5 above)
Comment (jimmywutangclan) [reply VolleyPaint/Assets/Scripts/Player/Shooting/PlayerShoot.cs]: This is old code that isn't going to be used, I will clean that up.

File: VolleyPaint/Assets/Blobcreate/Projectile Toolkit/Core/TrajectoryPredictor.cs (hunk 1 above)
Comment (jimmywutangclan) [reply VolleyPaint/Assets/Blobcreate/Projectile Toolkit/Core/TrajectoryPredictor.cs]: Not sure what happened tbh

File: VolleyPaint/Assets/Scripts/Ability/AbilityHolder.cs (hunk 2 above)
Comment (jimmywutangclan) [reply VolleyPaint/Assets/Scripts/Ability/AbilityHolder.cs]: I think we should merge Kim's branch when he's ready. Keeping this change for now shouldn't cause a merge conflict, right?

File: VolleyPaint/Assets/Scripts/Game/TeamAssignment.cs (hunk 3 above)
Comment (jimmywutangclan) [reply VolleyPaint/Assets/Scripts/Game/TeamAssignment.cs]: I think we should merge Kim's branch when he's ready. Keeping this change for now shouldn't cause a merge conflict, right?

**Review by slevenstein — APPROVED** [review]

**Pull request merged**